src/main/scala/com/fulcrumgenomics/basecalling/RunInfo.scala

```diff
@@ -0,0 +1,88 @@
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.fulcrumgenomics.basecalling
+
+import com.fulcrumgenomics.FgBioDef.FilePath
+import com.fulcrumgenomics.util.{Metric, ReadStructure, SampleBarcode, Template}
+import htsjdk.samtools.util.Iso8601Date
+
+/** Stores the result of parsing the run info file.
+  *
+  * @param run_barcode the unique identifier for the sequencing run and flowcell, stored as
+  *                   "<instrument-name>_<flowcell-barcode>".
+  * @param flowcell_barcode the flowcell barcode.
+  * @param instrument_name the instrument name.
+  * @param run_date the date of the sequencing run.
+  * @param read_structure the description of the logical structure of cycles within the sequencing run, including which cycles
+  *                      correspond to sample barcodes, molecular barcodes, template bases, and bases that should be skipped.
+  * @param num_lanes the number of lanes in the flowcell.
+  */
+case class RunInfo
+( run_barcode: String,
+  flowcell_barcode: String,
+  instrument_name: String,
+  run_date: Iso8601Date,
+  read_structure: ReadStructure,
+  num_lanes: Int
+) extends Metric
+
+object RunInfo {
+  /** Parses the run info file for the flowcell barcode, instrument name, run date, and read structure.
+    *
+    * @param runInfo the path to the RunInfo.xml file, typically in the run folder.
+    */
+  def apply(runInfo: FilePath): RunInfo = {
+    import scala.xml.XML
+    val xml = XML.loadFile(runInfo.toFile)
+    val flowcellBarcode = (xml \\ "RunInfo" \\ "Run" \\ "Flowcell").text
+    val instrumentName  = (xml \\ "RunInfo" \\ "Run" \\ "Instrument").text
+    val runDate         = (xml \\ "RunInfo" \\ "Run" \\ "Date").text
+    val segments        = (xml \\ "RunInfo" \\ "Run" \\ "Reads" \\ "Read").map { read =>
+      val isIndexedRead = (read \ "@IsIndexedRead").text.equals("Y")
+      val numCycles     = (read \ "@NumCycles").text.toInt
+      if (isIndexedRead) SampleBarcode(offset=0, length=numCycles) else Template(offset=0, length=numCycles)
+    }
+    val readStructure = new ReadStructure(segments, resetOffsets=true)
+    val numLanes = (xml \\ "RunInfo" \\ "Run" \\ "FlowcellLayout" \ "@LaneCount").text.toInt
+
+    RunInfo(
+      run_barcode      = s"${instrumentName}_$flowcellBarcode",
+      flowcell_barcode = flowcellBarcode,
+      instrument_name  = instrumentName,
+      run_date         = formatDate(runDate),
+      read_structure   = readStructure,
+      num_lanes        = numLanes
+    )
+  }
+
+  /** Formats the given date string.  The date string should be six or eight characters long with formats YYMMDD or
+    * YYYYMMDD respectively.
+    */
+  private def formatDate(date: String): Iso8601Date = {
```

Comment:
Technically this should be `parseDate` not `formatDate` since the latter implies going from `Date` -> `String`.